### PR TITLE
Refactor python extraction a bit for smaller functions

### PR DESCRIPTION
**Pull request by johnynek**

### CODE REVIEW

File: crates/driver/src/print_build.rs [hunk 1]
```diff
@@ -610,11 +615,14 @@ where
...
-                if let Some(ep) = &bin.binary_refs.target_value {
-                    k_strs.push(("binary_refs_value".to_string(), ep.to_string()));
+                if let Some(tv) = &bin.binary_refs.target_value {
+                    k_strs.push(("binary_refs_value".to_string(), tv.to_string()));
                 }
 
-                k_strs.push(("owning_library".to_string(), format!(":{}", lib_target.to_string())));
+                k_strs.push((
+                    "owning_library".to_string(),
+                    format!(":{}", lib_target),
```
Comment (johnynek) [inline crates/driver/src/print_build.rs:624]: don't need `lib_target.to_string()` here.

File: crates/driver/src/print_build.rs [hunk 2]
```diff
@@ -610,11 +615,14 @@ where
                     k_strs.push(("entity_path".to_string(), ep.to_string()));
                 }
 
-                if let Some(ep) = &bin.binary_refs.target_value {
-                    k_strs.push(("binary_refs_value".to_string(), ep.to_string()));
+                if let Some(tv) = &bin.binary_refs.target_value {
```
Comment (johnynek) [inline crates/driver/src/print_build.rs:618]: we copied `ep` but that seems to come from entity_path above.

File: crates/python_extractor/src/lib.rs [hunk 3]
```diff
@@ -71,7 +72,7 @@ lazy_static! {
     };
 }
 
-pub fn read_file_to_str(input_file: &PathBuf) -> Result<String, anyhow::Error> {
+pub fn read_file_to_str(input_file: &Path) -> Result<String, anyhow::Error> {
```
Comment (johnynek) [inline crates/python_extractor/src/lib.rs:75]: `&PathBuf` is an anti-pattern, use `&Path` which is a reference to an immutable Path.

File: crates/python_extractor/src/lib.rs [hunk 4]
```diff
@@ -120,68 +121,90 @@ pub fn read_file_to_str(input_file: &PathBuf) -> Result<String, anyhow::Error> {
     }
 }
 
-pub async fn extract_python(
-    relative_input_paths: String,
-    working_directory: PathBuf,
-    output: PathBuf,
-    label_or_repo_path: String,
+async fn extract_file(
```
Comment (johnynek) [inline crates/python_extractor/src/lib.rs:124]: write a function that just runs on a single file to reduce the size of the function below that calls it.

**Review by eed3si9n — APPROVED** [review]

File: crates/python_extractor/src/lib.rs [hunk 5]
```diff
@@ -120,68 +121,90 @@ pub fn read_file_to_str(input_file: &PathBuf) -> Result<String, anyhow::Error> {
...
-        } else {
-            defs.extend(expand_path_to_defs(&relative_path, file_p.as_ref()));
-        }
-        data_blocks.push(DataBlock {
-            entity_path: relative_path,
-            defs,
-            refs,
-            bzl_gen_build_commands,
-        })
+    for data_block in join_all(all_loads).await {
```
Comment (johnynek) [inline crates/python_extractor/src/lib.rs:202]: use join_all to run extraction in parallel rather than sequentially.

**Pull request merged**